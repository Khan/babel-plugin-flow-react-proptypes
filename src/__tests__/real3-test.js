const babel = require('babel-core');
const content = `
// @flow
import type { Job } from 'JLCommon';
export type JobViewImpression = Job & {
    resultId?: string,
    listIndex: number,
    searchType: ?string,
    viewSource: ?string,
};
`;

it('real3', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['@babel/env', '@babel/react', '@babel/flow'],
    plugins: [
      '@babel/syntax-flow',
      require('../'),
      "@babel/plugin-proposal-class-properties"
    ],
  }).code;
  expect(res).toMatchSnapshot();
});
