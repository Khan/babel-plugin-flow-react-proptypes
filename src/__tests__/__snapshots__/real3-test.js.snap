// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`real3 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
    value: true
});
exports.bpfrpt_proptype_JobViewImpression = undefined;

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var bpfrpt_proptype_JobViewImpression = {
    resultId: _propTypes2.default.string,
    listIndex: _propTypes2.default.number.isRequired,
    searchType: _propTypes2.default.string,
    viewSource: _propTypes2.default.string
};
exports.bpfrpt_proptype_JobViewImpression = bpfrpt_proptype_JobViewImpression;"
`;
